Feature common identhadler itf

mlohvynenko opened the pull request:
depends on https://github.com/aosedge/aos_core_lib_cpp/pull/347

mlohvynenko committed:
WIP: cmake: use dev repo

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>

mlohvynenko committed:
iam: identhandler: adapt ident handler interface changes

This patch adapts ident handler changes in the core lib library.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>

mlohvynenko committed:
common: iamclient: add public identity service handler

This patch adds a new public identity service handler
that implements ident handler interface.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>

MykolaSuperman commented on src/common/iamclient/publicidentityhandler.hpp:
```diff
@@ -0,0 +1,79 @@
+#define AOS_COMMON_IAMCLIENT_PUBLICIDENTITYHANDLER_HPP_
+
+#include <condition_variable>
+#include <future>
```
this header is necessary?

MykolaSuperman commented on src/common/iamclient/publicidentityhandler.hpp:
```diff
@@ -0,0 +1,79 @@
+#include <future>
+#include <mutex>
+#include <thread>
+#include <unordered_map>
```
ditto?

mlohvynenko committed:
cm: app: initialize public identity handler

This patch initializes the public identity handler
and all its dependencies in the communication
manager application

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>

MykolaSuperman reviewed (approved):
`Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>`

al1img commented on src/common/iamclient/publicidentityhandler.cpp:
```diff
@@ -0,0 +1,292 @@
+
+RetWithError<StaticString<cUnitModelLen>> PublicIdentityServiceHandler::GetUnitModel()
+{
+    LOG_INF() << "Get unit model";
```
ditto

al1img commented on src/common/iamclient/publicidentityhandler.cpp:
```diff
@@ -0,0 +1,292 @@
+{
+    std::lock_guard lock {mMutex};
+
+    LOG_INF() << "Subscribe to subjects changed";
```
dbg log

al1img commented on src/common/iamclient/publicidentityhandler.cpp:
```diff
@@ -0,0 +1,292 @@
+{
+    std::lock_guard lock {mMutex};
+
+    LOG_INF() << "Unsubscribe from subjects changed";
```
ditto

al1img commented on src/common/iamclient/publicidentityhandler.cpp:
```diff
@@ -0,0 +1,292 @@
+        ReceiveChangedSubjects();
+    }
+
+    LOG_INF() << "Public identity service handler thread stopped";
```
ditto
